### ISSUE TITLE
Style/HashExcept ignores negation

### ISSUE THREAD

**Issue by lobner** [rubocop/rubocop#13012]
## Expected behavior

I expect it to alter the code correctly.

In this case, it could be `slice` -- the following lines of code behave the same.

![Screenshot 2024-06-22 at 09 38 35](https://github.com/rubocop/rubocop/assets/1708153/7f30df7b-b6dd-455c-b09d-94ede70b2db0)

This is semantically the same.

Also, the Rails/NegateInclude - if run prior - would possibly solve it.
```ruby
test_rubocop.rb:3:18: C: [Correctable] Rails/NegateInclude: Use .exclude? and remove the negation part.
address_array = ![:street_name, :house_number, :floor, :zipcode, :city, :country].include?(k)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Actual behavior

The Style/HashExcept cop dismisses the `!` 

![image](https://github.com/rubocop/rubocop/assets/1708153/5eea9db6-bd58-4fd6-9795-9a8de49a1bab)

## Steps to reproduce the problem

Create a file `test_rubocop.rb` and put the following lines in it:

```ruby
➜ cat test_rubocop.rb
address_hash_1 = base_data.reject { |k, _v| ![:street_name, :house_number, :floor, :zipcode, :city, :country].include?(k) }
address_hash_2 = base_data.reject { |k, _v| [:street_name, :house_number, :floor, :zipcode, :city, :country].include?(k) }
```

Run: `rubocop test_rubocop.rb --only Style/HashExcept` and observe that it treats the lines as identical.

```ruby
➜ rubocop test_rubocop.rb --only Style/HashExcept
Inspecting 1 file
C

Offenses:

test_rubocop.rb:1:26: C: [Correctable] Style/HashExcept: Use except(:street_name, :house_number, :floor, :zipcode, :city, :country) instead.
address_hash_1 = base_data.reject { |k, _v| ![:street_name, :house_number, :floor, :zipcode, :city, :country].include?(k) }
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test_rubocop.rb:2:28: C: [Correctable] Style/HashExcept: Use except(:street_name, :house_number, :floor, :zipcode, :city, :country) instead.
address_hash_2 = base_data.reject { |k, _v| [:street_name, :house_number, :floor, :zipcode, :city, :country].include?(k) }
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

## RuboCop version

```
➜ rubocop -V
1.64.1 (using Parser 3.3.3.0, rubocop-ast 1.31.3, running on ruby 3.2.4) [arm64-darwin23]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-minitest 0.35.0
  - rubocop-performance 1.21.1
  - rubocop-rails 2.25.0
  - rubocop-rake 0.6.0
```
